### ISSUE TITLE
The new functions of follow_line and calibrate_white

### ISSUE THREAD

**Issue by JohnHZheng** [ev3dev/ev3dev-lang-python#670]
I am very eager of the new functions follow_line and calibrate_white in beta 5 code. 
However, the documentation of both are very limited. 

There is one sample code for follow_line, but I cannot make it work. It keeps complaining the robot is moving too fast even I set to speed to 5%. Or it will complain lost the line in the move. 

How do I use calibrate_white? Is it just run it once for an new environment and then it will work? Like to see some examples. 
Appreciate if anyone can share some knowledge.

**Comment by dwalton76** [comment]
For `follow_line` try using this program to fine tune the kp, ki, kd constants for your robot.  The ideal settings for these can vary a lot from robot to robot so they defaults may not work for yours.
https://github.com/ev3dev/ev3dev-lang-python/blob/ev3dev-stretch/utils/line-follower-find-kp-ki-kd.py

Regarding `calibrate_white` basically yes, just run it once when the environment changes (lighting, etc) and whatever the color_sensor is looking at when you run `calibrate_white` is what it will consider as `white` until you run it `calibrate_white` again.

**Comment by dwalton76** [comment]
FYI since these are both related to the python library https://github.com/ev3dev/ev3dev-lang-python/ would be a better place to open issues

**Comment by dlech** [comment]
issue has been transferred

**Comment by JohnHZheng** [comment]
We tried the find kpkikd in the [link](https://github.com/ev3dev/ev3dev-lang-python/blob/ev3dev-stretch/utils/line-follower-find-kp-ki-kd.py) and spent a lot of hours by tweaking the parameters. It never worked out for us. We get values like kp=4.7, ki=0.1, and kd=3. But it will always ends with lost the line exception no matter what tweaking we can make. We did print the PDF and use an oval for our testing field. 
Anyone has successful stories of make follow_line really working? Thanks
